packages/@tinacms/cli/src/cmds/init/detectEnvironment.ts

```diff
@@ -0,0 +1,236 @@
+    ),
+    name,
+    parentPath,
+    typescriptExists: false,
+    javascriptExists: false,
+    get resolve() {
+      return (typescript) =>
+        typescript
+          ? {
+              exists: this.typescriptExists,
+              path: this.fullPathTS,
+              parentPath: this.parentPath,
+            }
+          : {
+              exists: this.javascriptExists,
+              path: this.fullPathJS,
+              parentPath: this.parentPath,
+            }
+    },
+  }
+
+  result.typescriptExists = await fs.pathExists(result.fullPathTS)
+  result.javascriptExists = await fs.pathExists(result.fullPathJS)
+
+  return result
+}
+
+async function detectNextGlobalStyles(baseDir: string, usingSrc: boolean) {
+  let pathToGlobalStyles = ''
+  let globalStylesHasTailwind = false
+  let pathToApp = path.join(baseDir, usingSrc ? 'src' : 'pages', '_app')
+  if (fs.pathExistsSync(`${pathToApp}.js`)) {
+    pathToApp = `${pathToApp}.js`
+  } else if (fs.pathExistsSync(`${pathToApp}.tsx`)) {
+    pathToApp = `${pathToApp}.tsx`
+  } else {
+    pathToApp = ''
+  }
+  if (pathToApp) {
+    // read lines from file into array of strings
+    const lines = (await fs.readFile(pathToApp, 'utf8')).split('\n')
+    let stylesPath = ''
+    for (const line of lines) {
+      const match = line.match(/^import\s+["'](.*\.css)["'];?$/)
+      if (match) {
+        stylesPath = match[1]
+        break
+      }
+    }
+    if (stylesPath) {
+      if (stylesPath.startsWith('@')) {
+        stylesPath = path.join(baseDir, stylesPath.replace('@/', ''))
+      } else {
+        stylesPath = path.join(path.dirname(pathToApp), stylesPath)
+      }
+      pathToGlobalStyles = stylesPath
+
+      // compute path to styles file
+      if (fs.pathExistsSync(stylesPath)) {
+        // check if styles file imports tailwind
+        const globalStylesContent = await fs.readFile(
+          pathToGlobalStyles,
+          'utf8'
+        )
+        if (globalStylesContent.indexOf('@tailwind') !== -1) {
+          globalStylesHasTailwind = true
+        }
+      }
+    }
+  }
+  return { pathToGlobalStyles, globalStylesHasTailwind }
+}
+
+const detectEnvironment = async ({
+  baseDir = '',
+  pathToForestryConfig,
+  rootPath,
+  debug = false,
+}: {
+  baseDir?: string
+  pathToForestryConfig: string
+  rootPath: string
+  debug?: boolean
+}): Promise<InitEnvironment> => {
+  if (fs.pathExistsSync('.env.tina')) {
+    dotenv.config({ path: '.env.tina' })
+  }
+
+  // If there is a forestry config, ask user to migrate it to tina collections
+  const hasForestryConfig = await fs.pathExists(
+    path.join(pathToForestryConfig, '.forestry', 'settings.yml')
+  )
+  const hasTailwindConfig = await fs.pathExists(
+    path.join(baseDir, 'tailwind.config.js')
+  )
+  const sampleContentPath = path.join(
+    baseDir,
+    'content',
+    'posts',
+    'hello-world.md'
+  )
+  const usingSrc = !fs.pathExistsSync(path.join(baseDir, 'pages'))
```

Comment:
I wonder if we include app here as well since a lot of folks on nextJs are using the app router. 


EX:
```ts
// They are using `src` if the`src` folder exists and if  `src/app` exists or `src/pages` exists. 
const usingSrc = fs.pathExistsSync(path.join(baseDir, 'src')) && (fs.pathExistsSync(path.join(baseDir, 'src', 'app')) || fs.pathExistsSync(path.join(baseDir, 'src', 'pages'))  )  
```